x/dogfood/client/cli/query.go

```diff
@@ -60,3 +62,151 @@ func CmdQueryParams() *cobra.Command {
 
 	return cmd
 }
+
+func CmdQueryOptOutsToFinish() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "opt-outs-to-finish [epoch]",
+		Short: "shows the operator addresses whose opt out matures at the provided epoch",
+		Args:  cobra.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			clientCtx, err := client.GetClientQueryContext(cmd)
+			if err != nil {
+				return err
+			}
+			queryClient := types.NewQueryClient(clientCtx)
+			epoch := args[0]
+			cEpoch, err := strconv.ParseInt(epoch, 10, 64)
+			if err != nil {
+				return err
+			}
+			res, err := queryClient.OptOutsToFinish(
+				cmd.Context(), &types.QueryOptOutsToFinishRequest{Epoch: cEpoch},
+			)
+			if err != nil {
+				return err
+			}
+
+			return clientCtx.PrintProto(res)
+		},
+	}
+
+	flags.AddQueryFlagsToCmd(cmd)
+
+	return cmd
+}
+
+func CmdQueryOperatorOptOutFinishEpoch() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "operator-opt-out-finish-epoch [operator]",
+		Short: "shows the epoch at which an operator's opt out will be finished",
+		Args:  cobra.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			clientCtx, err := client.GetClientQueryContext(cmd)
+			if err != nil {
+				return err
+			}
+			queryClient := types.NewQueryClient(clientCtx)
+			operator := args[0]
+			res, err := queryClient.OperatorOptOutFinishEpoch(
+				cmd.Context(), &types.QueryOperatorOptOutFinishEpochRequest{
+					OperatorAccAddr: operator,
+				},
+			)
+			if err != nil {
+				return err
+			}
+
+			return clientCtx.PrintProto(res)
+		},
+	}
+
+	flags.AddQueryFlagsToCmd(cmd)
+
+	return cmd
+}
+
+func CmdQueryUndelegationsToMature() *cobra.Command {
```

Comment:
Should we still provide these functions in dogfood, seems not necessary here. All undelegations information should be retrieved from the `delegation` module, right?

Reply:
This function lists out the undelegations that will mature at epoch N. That information is not available with the delegation module.